enumerable.rb

```diff
@@ -0,0 +1,175 @@
+# rubocop:disable all
+module Enumerable
+  def my_each()
+    return to_enum(:my_each) unless block_given?
+
+    var = to_a
+    result = []
+    i = 0
+    while i < var.length
+      result << yield(var[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index()
+    return to_enum(:my_each_with_index) unless block_given?
+
+    var = to_a
+    i = 0
+    while i < var.length
+      result = yield(var[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    arr = []
+    my_each { |i| arr.push(i) if yield(i) }
+    arr
+  end
+
+  def my_all?(arg = nil)
+    var = to_a
+    result = true
+    var.my_each do |n|
+      if block_given?
+        if yield(n)
+          result = true
+        elsif !yield(n)
+          result = false
+        end
+      elsif arg.nil?
+        result = false if n.nil? || n == false
+      else
+        if arg === n
+          result = false
+          break
+        end
+      end
+    end
+    result
+  end
+
+  def my_any?(arg = nil)
+    var = to_a
+    result = true
+    var.my_each do |n|
+      if block_given?
+        if yield(n)
+          result = true
+        elsif !yield(n)
+          result = false
+        end
+      elsif arg.nil?
+        result = false if n.nil? || n == false
+      else
+        unless arg === n
+          result = false
+          break
+        end
+      end
+    end
+    result
+  end
```

Comment:
There are a couple of problems with this method:

![image](https://user-images.githubusercontent.com/2739245/94325065-4358ea80-ff62-11ea-8f60-df5eb3c09dcd.png)

You may use the following definitions to run this test:

```
true_array =  [nil, false, true, []]
```

![image](https://user-images.githubusercontent.com/2739245/94325115-79966a00-ff62-11ea-916b-a2355791ac77.png)

For this case, you may use:

```
words = %w[cat dog door rod blade]
```